src/site/content/en/blog/proxx-1/index.md

```diff
@@ -0,0 +1,553 @@
+
+Our critical CSS is already inlined thanks to CSS Modules and our Puppeteer-based prerenderer. For JavaScript we need to inline our critical modules _and their dependencies_. This task has varying difficulty, based on the bundler that you're using.
+
+{% Aside %}
+  **Note:** In this step we also subset our font files to contain only the glyphs that we need for our landing page. I am not going to go into detail on this step as it is not easily abstracted and sometimes not even practical. We still load the full font files lazily, but they are not needed for the initial render.
+{% endAside %}
+
+<figure class="w-figure w-figure--center">
+  <video controls muted
+    preload="metadata"
+    class="w-screenshot"
+    poster="https://storage.googleapis.com/web-dev-assets/js-heavy-sites/ssr-proxx-load-poster.jpg"
+      src="https://storage.googleapis.com/web-dev-assets/js-heavy-sites/ssr-proxx-load.mp4"
+      type="video/mp4; codecs=h264">
+  </video>
+  <figcaption class="w-figcaption w-figcaption--fullbleed">
+    With the inlining of our JavaScript we have reduced our TTI from 8.5s to 7.2s.
+  </figcaption>
+</figure>
+
+This shaved 1 second off our TTI. We have now reached the point where our `index.html` contains everything that is needed for the initial render and becoming interactive. The HTML can render while it is still downloading, creating our FMP. The moment the HTML is done parsing and executing, the app is interactive.
+
+## Aggressive code splitting
+
+Yes, our `index.html` contains everything that is needed to become interactive. But on closer inspection it turns out it also contains everything else. Our `index.html` is around 43 KB. Let's put that in relation to what the user can interact with at the start: We have a form to configure the game containing a couple of components, a start button and probably some code to persist and load user settings. That's pretty much it. 43 KB seems like a lot.
+
+<figure class="w-figure w-figure--center">
+  <picture>
+    <source srcset="proxx-loaded.webp" type="image/webp">
+    <img src="proxx-loaded.jpg">
+  </picture>
+  <figcaption class="w-figcaption">
+    The landing page of PROXX. Only critical components are used here.
+  </figcaption>
+</figure>
+
+To understand where our bundle size is coming from we can use a [source map explorer] or a similar tool to break down what the bundle consists of. As predicted, our bundle contains the game logic, the rendering engine, the win screen, the lose screen and a bunch of utilities. Only a small subset of these modules are needed for the landing page. Moving everything that is not strictly required for interactivity into a lazily-loaded module will decrease TTI _significantly_.
+
+<figure class="w-figure w-figure--center">
+  <style>
```

Comment:
I was wondering if it's possible to turn this into an image? My main concern is around inadvertently breaking the layout during some far future update to the site's design or CSS. In general, we've tried to discourage folks from doing inline code for this reason.

Reply:
Hm, I can turn it into a foreign object SVG. Let me try that.

Reply:
Works well. Done.